### PR TITLE
Fix Lock issue

### PR DESCRIPTION
**Pull request by vivsriaus**
Add a test for lock
display detailed error message when template deployment fails

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@vivsriaus__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (visriniv). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by vivsriaus** [comment]
@azuresdkci retest this please

**Comment by ogail** [comment]
@azuresdkci retest this please

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Policy/GetAzurePolicyAssignment.cs [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+    using System.Management.Automation;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Components;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Extensions;
+    using Newtonsoft.Json.Linq;
+
+    /// <summary>
+    /// Gets the policy assignment.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, "AzureRMPolicyAssignment", DefaultParameterSetName = GetAzurePolicyAssignmentCmdlet.ParameterlessSet), OutputType(typeof(PSObject))]
```
Comment (stankovski) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Policy/GetAzurePolicyAssignment.cs]: Name should be AzureRmXXX

File: src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Policy/NewAzurePolicyAssignment.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Entities.Resources;
+    using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Extensions;
+    using Microsoft.Azure.Common.Authentication;
+    using Microsoft.WindowsAzure.Commands.Utilities.Common;
+    using Newtonsoft.Json.Linq;
+
+    /// <summary>
+    /// Creates a policy assignment.
+    /// </summary>
+    [Cmdlet(VerbsCommon.New, "AzureRMPolicyAssignment"), OutputType(typeof(PSObject))]
```
Comment (stankovski) [inline src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Policy/NewAzurePolicyAssignment.cs]: Same as above

**Comment by vivsriaus** [comment]
@stankovski The policy files shouldn't have been included in this PR. I've deleted the files here, but will address your comments in my policy PR here: https://github.com/Azure/azure-powershell/pull/1014

**Pull request merged**